### PR TITLE
116 Webpage title

### PR DESCRIPTION
**Pull request by hamishtaplin**
### What is the context of this PR?

[Trello card](https://trello.com/c/blvgnemC/116-webpage-title-page-name-survey-name-dahl-3)

Adds a proper meta title to the page for display in browser tabs/history, etc.

##### Bonus content

Fixed issue with main content layer not being centered consistently. Added a `MainCanvas` component to control this layout.

### How to review 

- Navigate every page and check page title against the AC in the user story.
- Check layout now matches design

### CODE REVIEW

File: src/components/BaseLayout/BaseLayout.test.js [hunk 1]
```diff
@@ -6,20 +6,22 @@ let wrapper;
...
 describe("components/BaseLayout", () => {
   beforeEach(() => {
-    wrapper = shallow(<BaseLayout>Children</BaseLayout>);
+    wrapper = shallow(
+      <BaseLayout docTitle="BaseLayout Test">Children</BaseLayout>
+    );
   });
 
-  it("shoulder render", function() {
+  it("should render", function() {
```
Comment (samiwel) [inline src/components/BaseLayout/BaseLayout.test.js]: Haha, WTF?

File: src/containers/QuestionnaireDesignPage/QuestionnaireDesignPage.js [hunk 2]
```diff
@@ -42,6 +43,15 @@ export class QuestionnaireDesignPage extends Component {
     }
   };
 
+  getMetaTitle = () => {
+    const { questionnaire, page } = this.props;
+    let docTitle = `${questionnaire.title}`;
+    if (page.title) {
+      docTitle = `${page.title} - ${docTitle}`;
+    }
+    return docTitle;
```
Comment (WickyNilliams) [inline src/containers/QuestionnaireDesignPage/QuestionnaireDesignPage.js]: Use ternary?

```js
getMetaTitle = () => {
  const { questionnaire, page } = this.props;

  return page.title
    ? `${page.title} - ${questionnaire.title}`
    : `${questionnaire.title}`;
};
```

File: src/containers/QuestionnaireDesignPage/QuestionnaireDesignPage.js [hunk 3]
```diff
@@ -61,7 +71,11 @@ export class QuestionnaireDesignPage extends Component {
     }
 
     return (
-      <BaseLayout breadcrumb={breadcrumb} questionnaire={questionnaire}>
+      <BaseLayout
+        breadcrumb={breadcrumb}
+        questionnaire={questionnaire}
+        docTitle={`${this.getMetaTitle()}`}
```
Comment (WickyNilliams) [inline src/containers/QuestionnaireDesignPage/QuestionnaireDesignPage.js]: no need for template literal: ` docTitle={this.getMetaTitle()}`

File: src/containers/QuestionnaireDesignPage/QuestionnaireDesignPage.js [hunk 4]
```diff
@@ -72,14 +86,16 @@ export class QuestionnaireDesignPage extends Component {
...
           <Column gutters={false}>
-            <EditorSurface
-              section={section}
-              page={page}
-              onUpdatePage={onUpdatePage}
-              onUpdateSection={onUpdateSection}
-              focused={this.state.focused}
-              onFocus={this.handleFocus}
-            />
+            <MainCanvas>
```
Comment (WickyNilliams) [inline src/containers/QuestionnaireDesignPage/QuestionnaireDesignPage.js]: Why did you move this?

Comment (hamishtaplin) [reply src/containers/QuestionnaireDesignPage/QuestionnaireDesignPage.js]: To make it more consistent with the other pages. I think this bit of layout is a concern of the page template, not the component.

**Comment by hamishtaplin** [comment]
@samiwel It doesn't display the page title until you have given the page a name.

File: src/containers/QuestionnaireDesignPage/QuestionnaireDesignPage.js (hunk 4 above)
Comment (WickyNilliams) [reply src/containers/QuestionnaireDesignPage/QuestionnaireDesignPage.js]: Fair 👍

**Comment by WickyNilliams** [comment]
[WebPage Title: [Page Name] - [Survey Name] - Dahl (3)](https://trello.com/c/blvgnemC/116-webpage-title-page-name-survey-name-dahl-3)

**Comment by WickyNilliams** [comment]
What about favicon requirement?

**Comment by hamishtaplin** [comment]
@WickyNilliams DOH!

**Comment by samiwel** [comment]
@hamishtaplin Cool I see it now 👍
